[Profiler] Torch Profiler distributed info is not JSON serializable

sraikund16 opened the pull request:
Summary: To fix https://github.com/pytorch/pytorch/issues/133308 we must create an encoder for numpy values so we can serialize the distributed metadata to JSON.

Test Plan: Added unit test to check that numpy values can be serialized

Differential Revision: D62411619

aaronenyeshi reviewed (approved):
LGTM, pleas check CI signals!

albanD commented on docs/source/conf.py:
```diff
@@ -2097,6 +2097,7 @@
     "NoneType",
     "NoopLogger",
     "NumberType",
+    "NumpyEncoder",
```
Please do not add anything into this file.
You should either make this private (add an _ at the beginning) or properly document it.

albanD commented on torch/profiler/profiler.py:
```diff
@@ -32,10 +32,33 @@
     "tensorboard_trace_handler",
     "profile",
     "ExecutionTraceObserver",
+    "NumpyEncoder",
```
You can remove this entry now that it is not public anymore.

sraikund16 replied:
removed!

albanD reviewed (approved):
SGTM from the public API point of view!

sraikund16 committed:
[Profiler] Torch Profiler distributed info is not JSON serializable (#135548)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/135548

To fix https://github.com/pytorch/pytorch/issues/133308 we must create an encoder for numpy values so we can serialize the distributed metadata to JSON.

Test Plan: Added unit test to check that numpy values can be serialized

Reviewed By: aaronenyeshi

Differential Revision: D62411619
